### ISSUE TITLE
Images Aligned with Classes and Enhanced Image plugin as non-default

### ISSUE THREAD

**Issue by MrKarlDilkington** [ExchangeCore/Concrete5-CKEditor#28]
This issue is based on the discussion here: #27

Because the Enhanced Image plugin does not have margin controls, I think it might be better to remove it as a default plugin and instead make it optional (the default image function does have margin controls).

"Images Aligned with Classes"
http://ckeditor.com/tmp/4.4.0/widget-styles.html

I think adding this would be very useful. It would provide styling class hooks for images and captions.

It would disable the inline CSS floats, which would require CSS added to a theme. I don't believe this is an issue because the use of captions requires adding CSS to a theme also. These could be done at the same time. Example CSS for adding caption and margin styling could be made available for users in the add-on documentation.

"CKEditor 4.4 introduces the new config.image2_alignClasses configuration option, which, along with the config.image2_captionedClass setting, gives developers more control over the widgets and makes the styling process much easier. For example, the following configuration:

```
CKEDITOR.replace( 'editor1', {
    extraPlugins: 'image2',
    image2_alignClasses: [ 'image-left', 'image-center', 'image-right' ],
    image2_captionedClass: 'image-captioned',
} );
```

will produce class-driven, stylable markup (e.g. class="image-right" instead of style="float:right"), avoiding inflexible and non-semantic inline CSS:

```
<figure class="image-captioned image-right">
    <img alt="Image" src="example.png" />
    <figcaption>Caption</figcaption>
</figure>
```

"

**Comment by joe-meyer** [comment]
@MrKarlDilkington I think the main reason we decided to use the enhanced image was becuase it gave the ability to justify the image in an easy user interface. It's also a far simpler interface so I feel like we should leave it at that and if someone needs the more advanced features, they can go in and disable the plugin (not sure what happens to things like the caption when that happens). Thoughts?

**Comment by MrKarlDilkington** [comment]
@joe-meyer 

I agree that the Enhanced Image plugin is easier to use than the default image plugin (I am thinking specifically about the image resizer). 

Because the Enhanced Image plugin does not include default margin and caption styling, I think this will cause confusion. Users will try to use the caption feature and think it is broken because it doesn't appear to do anything. Users will also be confused because there is no margin input option (it only supplies an inline float tag).

For these reasons, I think the Enhanced Image plugin should not be default. Instead it could be the "advanced" plugin, since it needs CSS to work (for caption and margin). As the advanced plugin, we can use the Image Aligned with Classes feature to provide more useful selector hooks for styling.

I am more than willing to provide basic boilerplate CSS to use the plugin to get users up and running quickly.

**Comment by joe-meyer** [comment]
Fixed via 1fdfb5ab7808a7bc2e3ddffa05a1fb182809c068

**Comment by joe-meyer** [comment]
Prematurely closed.... I did change the default image plugin to not be enhanced images, I did not add the image2_\* options to the config

**Comment by joe-meyer** [comment]
Closed via #37 Thanks @MrKarlDilkington
